Further tuning of the CORE_ROOT folder content

trylek opened the pull request:
Based on Sergey's feedback I'm putting back the alternate JITs
(linuxonjit, protononjit) as they are actively used by the JIT team
in specialized scenarios. I have also removed the PDB subfolder -
while we can have an orthogonal discussion what PDB's we may want
to make available to test runs in Helix, the current subfolder has
no value and was only present due to the pre-existing trivial script
logic blindly copying over the entire binary artifact subtree.

Thanks

Tomas

trylek committed:
Further tuning of the CORE_ROOT folder

Based on Sergey's feedback I'm putting back the alternate JITs
(linuxonjit, protononjit) as they are actively used by the JIT team
in specialized scenarios. I have also removed the PDB subfolder -
while we can have an orthogonal discussion what PDB's we may want
to make available to test runs in Helix, the current subfolder has
no value and was only present due to the pre-existing trivial script
logic blindly copying over the entire binary artifact subtree.

Thanks

Tomas

sandreenko reviewed (approved):
LGTM, thank you
cc @dotnet/jit-contrib

jkoritzinsky commented on src/coreclr/tests/src/Common/Directory.Build.targets:
```diff
@@ -30,19 +30,14 @@
       <RunTimeArtifactsIncludeFolders Include="/" />
       <RunTimeArtifactsIncludeFolders Include="IL/" />
       <RunTimeArtifactsIncludeFolders Include="crossgen2/" />
-      <RunTimeArtifactsIncludeFolders Include="PDB/" />
```
The PDB folder is how we get stack traces for runtime crashes. We shouldn’t remove it (or we should go straight from including the folder wholesale to including specific files in the folder)

trylek replied:
Hmm, can you please explain to me how Helix test executions locate the folder?

jkoritzinsky replied:
I don’t remember exactly how, but we pass information somewhere in an environment variable so Watson can find the PDBs.

echesakovMSFT replied:
> Hmm, can you please explain to me how Helix test executions locate the folder?

It here https://github.com/dotnet/runtime/blob/d68342ceeb69adff35d5b3462091d339a700c1a5/src/coreclr/tests/helixpublishwitharcade.proj#L196

trylek replied:
Thanks Jeremy and Egor for your feedback. I have put back the PDB folder and just added explicit exclusions for files that don't seem related to runtime call stacks. Please let me know if you believe any of the files I'm removing need to be kept. This still reduces the folder by about 175 MB.

trylek committed:
Replace complete PDB removal with targeted exclusions per PR feedback

jkoritzinsky commented on src/coreclr/tests/src/Common/Directory.Build.targets:
```diff
@@ -30,8 +30,20 @@
+      <RunTimeArtifactsExcludeFiles Include="PDB/ildasm.pdb" />
+      <RunTimeArtifactsExcludeFiles Include="PDB/linuxonjit.pdb" />
+      <RunTimeArtifactsExcludeFiles Include="PDB/mcs.pdb" />
+      <RunTimeArtifactsExcludeFiles Include="PDB/mscordaccore.pdb" />
```
Can we drop the DAC/DBI symbols and still get SOS working on managed dumps from CI?

cc: @hoyosjs

jkoritzinsky replied:
I think we should keep the DAC/DBI symbols just in case. The rest of the ones listed here we can drop IMO.

hoyosjs replied:
The DAC/DBI pdb's are not needed. The debuggee will never have any of their frames in the callstack (only a debugger is expected to have such frames).

BruceForstall replied:
I don't think you should exclude ilasm/ildasm -- they are product binaries.

So what is affected by this? E.g., what can I not do with the exclusion to the non-default JIT and SuperPMI ones?

BruceForstall replied:
E.g., superpmi.pdb and superpmi-shim-collector.pdb are PDBs associated with EXEs that run during the pri-0 test run. So will we not get core dumps if they fail?

trylek replied:
I have double-checked with @noahfalk that PDB's are not needed for producing crash dumps. For symbolic Watson traces they probably would be necessary (adding another 30 MB to the correlation payload).

jashook replied:
Do we have any testing that would use ilasm/ildasm?

jashook replied:
I think though that I agree that the pruning should be distinguished between the ci and the dev loop. Seems like we can be more careless with size in the core_root for dev usage.

Although it would be nice to come to a minimal set of requirements for our core_root for both ci and dev.

BruceForstall replied:
> Do we have any testing that would use ilasm/ildasm?

Unsure, but if not, we should

We used to have ildasm/ilasm round-trip testing, but I think we've lost it.

jashook replied:
Right, this was lost in the move to azdo.

jashook replied:
Can I suggest that we maybe have a flag `MinimalCoreRoot`. Which we can set in the CI. We have no reason to include ildasm/ilasm in our helix submissions, and the day we re-create the il* roundtrip tests those can include the ilasm/ildasm binaries and pdbs.

trylek committed:
Retain PDB's apparently related to DBI / DAC per Jeremy's feedback

trylek committed:
Put back exclusions for DAC / DBI per Juan's feedback

trylek committed:
Make ilasm.pdb / ildasm conditional; put back x64 / x86

jashook commented on src/coreclr/tests/src/Common/Directory.Build.targets:
```diff
@@ -29,16 +29,49 @@
       <RunTimeArtifactsIncludeFolders Include="R2RTest/" />
 
+      <!-- Used for ARM / ARM64 cross-compilation -->
+      <RunTimeArtifactsIncludeFolders Include="x86/" />
```
I have a feeling we should use the bin folder for these cross arch crossgen usages and keep these removed. @echesakovMSFT opinions?

echesakovMSFT replied:
I don't have strong opinion here - I am fine with not including these in Core_Root as long as crossgen_comparison tests continue working.

jashook replied:
crossgen comparison already uses the bin directory to my knowledge. There was something that broke with the removal. @BruceForstall what was that?

jashook replied:
I found it: https://github.com/dotnet/runtime/pull/35243#issuecomment-620407386

jashook replied:
https://github.com/dotnet/runtime/pull/35577 addresses this issue and I believe the only usage.

jashook reviewed (approved):
Thank you for these changes @trylek!

trylek committed:
Remove the x64 / x86 again

Jarret is fixing cross-compilation in a separate PR so we no longer
need to put these folders back to CORE_ROOT.

Thanks

Tomas

Jarret's PR for reference: https://github.com/dotnet/runtime/pull/35577
